### DISCUSSION TITLE
Subdomains and Cookies

### DISCUSSION THREAD

**Discussion by imphillipzissou** [payloadcms/payload#1193] [Q&A] [answered]
Hey folks,

Running into an issue where Payload running on a subdomain fails to set an auth cookie properly.

With Payload is running on `https://payload.domain.com` and a React frontend is running on `https://domain.com`.

User collection:
```
import { CollectionConfig } from 'payload/types';
export const User : CollectionConfig = {
  slug: 'user',
  auth: {
    cookies: {
      secure: true, // domain.com is secure (https)
      domain: `.domain.com`,
      sameSite: 'none' // Tried with all options, strict, lax, none
    },
  },
  access: {
    // ...
  },
  fields: [
    // ...
  ]
};
```

From inspecting requests, the correct response appears to be sent from payload however no cookie is ever actually set.

![Screen Shot 2022-10-02 at 8 22 16 AM](https://user-images.githubusercontent.com/7230065/193453887-c3e4f371-906c-43a7-9288-405d881a676a.png)

![Screen Shot 2022-10-02 at 8 31 39 AM](https://user-images.githubusercontent.com/7230065/193454068-c1e329ef-8d8c-4e2d-878f-e9730bd08491.png)

The issue seems similar to [discussion #401](https://github.com/payloadcms/payload/discussions/401#discussion-3767029) however I wasn't able to identify the issue with the cookie not setting.

Any thoughts?

**Comment by jmikrut** [comment]
Hey @imphillipzissou — this is strange. At first glance everything looks right. Can you try this in Chrome? I believe Chrome will show a little yellow warning sign if a cookie is not set successfully. And it typically comes with the reason why. Looks like you're using Safari, right?

**Comment by imphillipzissou** [comment]
@jmikrut Thanks for replying so quickly! Correct, this was with Safari. I just tried this on Chrome too, however same behavior and no little yellow warnings. 

![Screen Shot 2022-10-02 at 10 01 13 AM](https://user-images.githubusercontent.com/7230065/193458116-c6f6b07d-5ebc-449f-a29a-7b2cf64cf2f9.png)

![Screen Shot 2022-10-02 at 10 00 14 AM](https://user-images.githubusercontent.com/7230065/193458118-924e6847-eef5-48f7-a7cd-9742c69de691.png)

**Comment by jmikrut** [comment]
Hmm, the only thing I can see here is that in Chrome, the `SameSite` property appears to be set to `Strict`. Are you specifying `SameSite` to `None`?

**Reply by imphillipzissou** [reply]
Apologies for changing it in the screenshot. I've attempted it with all the options (`lax`, `strict`, and `none`) in Chrome. 

Not sure if it would make a difference but I am attempting this over the REST `/api/[colleciton-slug]/login` endpoint. Working on reproducing the issue isolated from my project.

**Comment by imphillipzissou** [comment, marked as answer]
Took a minute but here's what I did to reproduce.

1. Cloned and installed the Payload Next Auth repo: [payload-cms-next](https://github.com/payloadcms/payload-cms-next). Locally, I'm using Traefik, dnsmasq, for the certificate and resolution. Configured this to run and resolve on `payload.cookie.domain.com`.

2. Cloned and installed the Next Auth Frontend repo: [next-auth-frontend](https://github.com/payloadcms/next-auth-frontend). Again, locally using Traefik and dnsmasq for valid certs and resolution. Configured this to run and resolve on `cookie.domain.com`. Set the `NEXT_PUBLIC_CMS_URL` env. var. to point to `payload.cookie.domain.com`.

Then I tried a few different approaches.

1. Did **not** modify the `Users` collection to observe the initial behavior:
- Logging into payload at directly at `payload.cookie.domain.com` works as expected, sets a cookie for the payload subdomain.
- Logging into next-auth at `cookie.domain.com` also works until the page is refreshed. No cookie was set though, as expected (different domain).

2. Did modify the `Users` collection:
```
// ...
auth : {
  cookies: {
      secure: true,
      sameSite: 'strict', // Also tried `lax`, and `none`
      domain: '.cookie.domain.com' // Also tried `cookie.domain.com` and `.domain.com` to rule out it being the naked domain.
  },
  // ...
}
```
- Logging into payload works (as expected).
- Logging into next-auth at `cookie.domain.com` works until page is refreshed. **However no cookie is set though.**
- The console and networking logs show a similar response to my initial post where the response contains a `Set-Cookie` to the domain `cookie.domain.com`. However, again, doesn't actually set the cookie.

Additional (kind of) related behaviors:
- The logout button in next-auth says the 'You are already logged out.` However, I can still click `Account` up top and return to the Account page. Appears the only way to logout is to refresh the page.

Not sure what the heck is going on here! lol 

Edit: Seems like it's specific to REST fetch requests so looking into this and CORS configuration a little more.

Edit: Am a goof. lol There are a couple things at play. Ultimately I was calling the payload rest api in `getServerSideProps`. Which is obviously not from the browser. So the request wouldn't contain the cookie. What I ultimately did was read the httponly cookie in `getServerSideProps` then set it as an `Authorization: JWT {cookie token}` on the payload request.
